src/test/java/com/aws/greengrass/deployment/ConnectivityValidatorTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.deployment;
+
+import com.aws.greengrass.config.Topic;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.dependency.Context;
+import com.aws.greengrass.deployment.model.DeploymentResult;
+import com.aws.greengrass.lifecyclemanager.Kernel;
+import com.aws.greengrass.mqttclient.MqttClient;
+import com.aws.greengrass.testcommons.testutilities.GGExtension;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.crt.mqtt.MqttClientConnection;
+import software.amazon.awssdk.iot.AwsIotMqttConnectionBuilder;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import software.amazon.awssdk.services.greengrassv2data.model.ListThingGroupsForCoreDeviceRequest;
+
+import java.util.HashMap;
+import java.util.concurrent.CompletableFuture;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({GGExtension.class, MockitoExtension.class})
+public class ConnectivityValidatorTest {
+    @Mock
+    private MqttClient mqttClient;
+    @Mock
+    private AwsIotMqttConnectionBuilder awsIotMqttConnectionBuilder;
+    @Mock
+    private MqttClientConnection mqttClientConnection;
+    @Mock
+    private GreengrassServiceClientFactory factory;
+    @Mock
+    private GreengrassV2DataClient greengrassV2DataClient;
+    @Mock
+    private DeviceConfiguration deviceConfiguration;
+    @Mock
+    private DeviceConfiguration deploymentConfiguration;
+
+    private Kernel kernel;
+    private final Context context = new Context();
+
+    Topics topics = Topics.of(context, "topics-name", null);
+    Topic topic = Topic.of(context, "topic-name", "value");
+    Topic differentTopic = Topic.of(context, "topic-name", "different-value");
+    ConnectivityValidator validator;
+
+
+    @BeforeEach
+    void beforeEach() {
+        kernel = new Kernel();
```

Comment:
close the kernel.

Use the Context from the kernel here, don't make your own Context